File: Python/03_storage_json.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os 
from couchbase import Couchbase
from blessings import Terminal
from pprint import pprint
import json
import hashlib

t = Terminal()

os.system('clear')
print t.bold_red("--------------------------------------------------------------------------")
print t.bold_red("Couchbase JSON Document Storage Operations")
print t.bold_red("--------------------------------------------------------------------------")
print

# connect to the default bucket
cb = Couchbase.connect(bucket='default')

# Create a new dictionary
mydoc = \
{
   "doctype": "test",
   "name": "John Smith"
}

print t.bold("Set a Document")
print

# store a json doc (automatically converted from dictionary to JSON)
print cb.set("mydoc", mydoc)
print

print t.bold("Get a Document")
print

# retrieve, get type of value, and attempt parse of json if string
val = cb.get("mydoc")

print t.bold("Pretty print a document")
print

# pretty print the document
pprint(val.value, indent=4)
print

print t.bold("Print the document as JSON")
print

print json.dumps(val.value)
print

print t.bold("Delete the document")
print

# delete the test docs to repeat tests 
cb.delete("mydoc")

print
print t.bold_red("--------------------------------------------------------------------------")
print

File: Nodejs/03_storage_json.js
var Connection = require('couchbase').Connection;

console.log("--------------------------------------------------------------------------");
console.log("Couchbase JSON Doc Storage");
console.log("--------------------------------------------------------------------------");

var cb = new Connection({
	"password": "",
    "host": "localhost",
	"bucket": "default"
}, function(err) {
	if (err) {
		throw (err)
	}


	var myDoc = {
		doctype : "test",
	   	name : "John Smith"
	};
	
	console.log("Set Document");
	cb.set("mydoc", myDoc, function(err, result) {
		// in the call back
		cb.get("mydoc", function(err, result) {
			console.log('mydoc.name : '+ result.value.name);

			cb.touch("mydoc", 10, function(err, result) {
				console.log(" Touch the doc with TTL");
				console.log("\n--------------------------------------------------------------------------");
				process.exit(0);
			});


		});
	});




});

